src/pydicom/dataset.py

```diff
@@ -1822,6 +1823,8 @@ def compress(
         decoding_plugin: str = "",
         encapsulate_ext: bool = False,
         jls_error: int | None = None,
```

Comment:
I didn't think of this before, but we had the discussion about keyword-only somewhere ... I think it should apply at least to the various jxxx params in `compress` also, i.e. insert a '*' in the function definition - much easier to change arg names later, not possible for user to get order wrong.

Reply:
Yeah, I'll do something like that when I do the decompress()/compress() PR. I'm trying to avoid doing too much in one PR.

Reply:
Done